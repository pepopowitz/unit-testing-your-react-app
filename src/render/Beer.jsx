import React from 'react';

import Brewery from './Brewery';
import BeerStyle from './BeerStyle';

export default function Beer({ beer }) {
    return (
        <article className="beer">
            <h1>
                {beer.name}
            </h1>

            <Brewery brewery={beer.brewery} />
            <BeerStyle beerStyle={beer.beerStyle} />
        </article>
    );
}
