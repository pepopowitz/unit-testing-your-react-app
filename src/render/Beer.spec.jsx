import Beer from './Beer';
import React from 'react';

import { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

it('renders a beer', () => {
    const beer = {
        name: 'Mudpuppy Porter'
    };

    const wrapper = shallow(<Beer beer={beer} />);

    expect(wrapper.find('.beer')).toHaveLength(1);
});

it('doesn`t shallow render brewery', () => {
    const beer = {
        name: 'Mudpuppy Porter',
        brewery: 'Central Waters'
    };

    const wrapper = shallow(<Beer beer={beer} />);

    expect(wrapper.find('.brewery')).toHaveLength(0);
});

it('renders brewery', () => {
    const beer = {
        name: 'Mudpuppy Porter',
        brewery: 'Central Waters'
    };

    const wrapper = render(<Beer beer={beer} />);

    expect(wrapper.find('.brewery')).toHaveLength(1);

});

it('deep renders brewery', () => {
    const beer = {
        name: 'Mudpuppy Porter',
        brewery: 'Central Waters'
    };

    const wrapper = mount(<Beer beer={beer} />);

    expect(wrapper.find('.brewery')).toHaveLength(1);
});

it('snapshots a beer with shallow', () => {
    const beer = {
        name: 'Mudpuppy Porter',
        brewery: 'Central Waters'
    };

    const wrapper = shallow(<Beer beer={beer} />);

    expect(toJson(wrapper)).toMatchSnapshot();
});

it('snapshots a beer with render', () => {
    const beer = {
        name: 'Mudpuppy Porter',
        brewery: 'Central Waters'
    };

    const wrapper = render(<Beer beer={beer} />);

    expect(toJson(wrapper)).toMatchSnapshot();
});

it('snapshots a beer with mount', () => {
    const beer = {
        name: 'Mudpuppy Porter',
        brewery: 'Central Waters'
    };

    const wrapper = mount(<Beer beer={beer} />);

    expect(toJson(wrapper)).toMatchSnapshot();
});
