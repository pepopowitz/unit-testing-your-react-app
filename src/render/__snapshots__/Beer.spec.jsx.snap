// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots a beer with mount 1`] = `
<Beer
  beer={
    Object {
      "brewery": "Central Waters",
      "name": "Mudpuppy Porter",
    }
  }
>
  <article
    className="beer"
  >
    <h1>
      Mudpuppy Porter
    </h1>
    <Brewery
      brewery="Central Waters"
    >
      <h2
        className="brewery"
      >
        Central Waters
      </h2>
    </Brewery>
    <BeerStyle />
  </article>
</Beer>
`;

exports[`snapshots a beer with render 1`] = `
<article
  class="beer"
>
  <h1>
    Mudpuppy Porter
  </h1>
  <h2
    class="brewery"
  >
    Central Waters
  </h2>
</article>
`;

exports[`snapshots a beer with shallow 1`] = `
<article
  className="beer"
>
  <h1>
    Mudpuppy Porter
  </h1>
  <Brewery
    brewery="Central Waters"
  />
  <BeerStyle />
</article>
`;
